modules/packages/msiexec-list.vbs

```diff
@@ -0,0 +1,47 @@
+' Helper script to enumerate installed software (packages)
+' Based on original work of Cory Coager of CDPHP
+' This module reads the installed packages from the registry using WMI
+
+Const HKLM = &H80000002 'HKEY_LOCAL_MACHINE
+strComputer = "."
+strKeys=Array("SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\", "SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall\")
+strDisplayNameEntryA = "DisplayName"
+strDisplayNameEntryB = "QuietDisplayName"
+strInstallDateEntry = "InstallDate"
+strDisplayVersionEntry = "DisplayVersion"
+
+' Read registry using WMI
+Set objReg = GetObject("winmgmts://" & strComputer & "/root/default:StdRegProv")
+' Iterate through each uninstall registry key
+For Each strKey in strKeys
+  ' Enumerate the subkeys
+  objReg.EnumKey HKLM, strKey, arrSubkeys
+  ' Iterate through each subkey
+  For Each strSubkey In arrSubkeys
+    intRet1 = objReg.GetStringValue(HKLM, strKey & strSubkey, strDisplayNameEntryA , strDisplayNameValue)
+    ' Check return value
+    If intRet1 <> 0 Then
+      ' Read the Displayname
+      objReg.GetStringValue HKLM, strKey & strSubkey, strDisplayNameEntryB, strDisplayNameValue

```

Comment:
would you want to handle/notify about errors on the GetStringValue calls from this line 25 down? What will happen if they fail? Just missing/empty strings for those values?

Reply:
Good point! However I believe "version" and "install date" are optional - we shouldn't fail if they're not set in registry.

[The GetStringValue docs][d] don't specify what happens to the output argument in case of error. So let's set all variables to empty strings at the beginning of each loop iteration.

[d]: https://docs.microsoft.com/en-us/previous-versions/windows/desktop/regprov/getstringvalue-method-in-class-stdregprov

Regarding this particular line - I believe it was assumed that in case of error the output variable "strDisplayNameValue" is set to empty string, since this check is performed afterwards. But since it's not documented, let's add an explicit error checking here, and for optional values follow the pattern of "using whatever GetStringValue set this variable to, even if it returned with error"